test/realm_test.dart

```diff
@@ -715,6 +714,169 @@ Future<void> main([List<String>? args]) async {
+    realm.close();
+  });
+
+  test('Realm open async for local configuration', () async {
+    final configuration = Configuration.local([Car.schema]);
+    final realm = await Realm.open(configuration);
+    expect(realm.isClosed, false);
+    realm.close();
+  });
+
+  baasTest('Realm open async and get progress', (appConfiguration) async {
+    final app = App(appConfiguration);
+    final credentials = Credentials.anonymous();
+    final user = await app.logIn(credentials);
+    final configuration = Configuration.flexibleSync(user, [Task.schema]);
+
+    final realm = await Realm.open(configuration, onProgressCallback: (transferredBytes, totalBytes) {
+      print("transferredBytes: $transferredBytes, totalBytes:$totalBytes");
+    });
+    expect(realm.isClosed, false);
+    realm.close();
+  });
+
+  baasTest('Realm open async, add data and get progress', (appConfiguration) async {
+    final app = App(appConfiguration);
+
+    final user1 = await app.logIn(Credentials.anonymous(reuseCredentials: false));
+    final configuration1 = Configuration.flexibleSync(user1, [Task.schema]);
+    final realm1 = Realm(configuration1);
+    realm1.subscriptions.update((mutableSubscriptions) => mutableSubscriptions.add(realm1.all<Task>()));
+    realm1.close();
+
+    final user2 = await app.logIn(Credentials.anonymous(reuseCredentials: false));
+    final configuration2 = Configuration.flexibleSync(user2, [Task.schema]);
+    final realm2 = Realm(configuration2);
+    realm2.subscriptions.update((mutableSubscriptions) => mutableSubscriptions.add(realm2.all<Task>()));
+    realm2.write(() {
+      for (var i = 0; i < 100; i++) {
+        realm2.add(Task(ObjectId()));
+      }
+    });
+    realm2.close();
+
+    final realmAsync1 = Realm.open(configuration1, onProgressCallback: (transferredBytes, totalBytes) {
+      print("transferredBytes: $transferredBytes, totalBytes:$totalBytes");
+    });
+    var syncedRealm = await realmAsync1;
+    expect(syncedRealm.isClosed, false);
+    syncedRealm.close();
+  });
+
+  baasTest('Realm open async and cancel for local flexibleSync', (appConfiguration) async {
+    final app = App(appConfiguration);
+    final credentials = Credentials.anonymous();
+    final user = await app.logIn(credentials);
+    final configuration = Configuration.flexibleSync(user, [Task.schema]);
+
+    var cancellationToken = CancellationToken();
+    final realm = Realm.open(configuration, cancellationToken: cancellationToken);
+    cancellationToken.cancel();
+    expect(() async => await realm, throws<CancelledException>());
+  });
+
+  test('Realm open async and cancel for local configuration', () async {
+    final configuration = Configuration.local([Car.schema]);
+    var cancellationToken = CancellationToken();
+    final realm = Realm.open(configuration, cancellationToken: cancellationToken);
+    cancellationToken.cancel();
+    var openedRealm = await realm; // You can't cancel local Realm
```

Comment:
Do you think it's okay to have different behavior for local realms? The Realm is returned, although the opening is cancelled.